src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -591,4 +565,69 @@ private static string GetProductInfo()
 
         return info.ToString();
     }
+
+    private static bool ValidateArguments(string[] args, CommandLineApplication app)
+    {
+        static HashSet<string> GetValidArguments(CommandLineApplication app)
+        {
+            var validArguments = new HashSet<string>();
+            foreach (var option in app.GetOptions())
+            {
+                if (!string.IsNullOrEmpty(option.LongName))
+                {
+                    validArguments.Add(option.LongName);
+                }
+                if (!string.IsNullOrEmpty(option.ShortName))
+                {
+                    validArguments.Add(option.ShortName);
+                }
+            }
+
+            return validArguments;
+        }
```

Comment:
```suggestion
        static HashSet<ReadOnlyMemory<char>> GetValidArguments(CommandLineApplication app)
        {
            HashSet<ReadOnlyMemory<char>> validArguments = new(new MemoryContentsComparer<char>());
            foreach (var option in app.GetOptions())
            {
                if (!string.IsNullOrEmpty(option.LongName))
                {
                    validArguments.Add(option.LongName.AsMemory());
                }
                if (!string.IsNullOrEmpty(option.ShortName))
                {
                    validArguments.Add(option.ShortName.AsMemory());
                }
            }

            return validArguments;
        }
```

Reply:
Interesing that solving previous comment I did apply those same changes you are suggesting
Except they way you are initializing the hash

Your suggestion
```
HashSet<ReadOnlyMemory<char>> validArguments = new(new MemoryContentsComparer<char>());
```

My code 
```
HashSet<ReadOnlyMemory<char>> validArguments = new HashSet<ReadOnlyMemory<char>>(new MemoryContentsComparer<char>());
```


I changed it to use your way to initialize it (simpler)